src/main/kotlin/com/uber/nanoscope/Main.kt

```diff
@@ -143,30 +147,61 @@ class StartHandler(private val args: List<String>): VersionedHandler() {
         trace.stop()
     }
 
-    private fun getPackageName(): String? {
+    private fun validParams(): Boolean {
         if (args.isEmpty()) {
-            return null
+            return false
         }
 
-        val usage = "usage: nanoscope start [--package=com.example]"
-        if (args.size != 1) {
+        if (args.size > max_params) {
             println(usage)
             exitProcess(1)
         }
+        
+        return true;
+    }
 
-        val parts = args[0].split('=')
-        if (parts.size != 2) {
-            println(usage)
-            exitProcess(1)
+    private fun getPackageName(): String? {
+        if (!validParams()) {
+            return null
         }
 
-        if (parts[0] != "--package") {
-            println(usage)
-            exitProcess(1)
+        for (i in 0..args.size-1) {
+            val parts = args[i].split('=')
+            if (parts.size == 2 && parts[0] == "--package") {
+               return parts[1]
+            } // else continue
```

Comment:
This else condition used to show a usage message. Let's keep that behavior.

Reply:
I wanted to allow specification of options in arbitrary order (hence the loop to inspect both arguments). Considering this, we can't error out here in case it's the other option that is being currently processed.